cli/build.rs

```diff
@@ -1,28 +1,35 @@
 //! Build script to extract git hash of iroha build and to check runtime validator
 
-use eyre::{eyre, Result};
-use vergen::{vergen, Config};
+use std::error::Error;
 
 const DEFAULT_VALIDATOR_PATH: &str = "../default_validator";
 
-fn main() -> Result<()> {
+fn main() -> Result<(), Box<dyn Error>> {
     println!("cargo:rerun-if-changed=build.rs");
     println!("cargo:rerun-if-changed={}", DEFAULT_VALIDATOR_PATH);
 
     extract_git_hash()?;
-    check_default_validator()
+
+    // HACK: used by Nix, since at the moment
+    // the checks are a process that's hard to accomodate
+    // in Nix environment
+    if std::option_env!("IROHA_SKIP_WASM_CHECKS").is_none() {
+        check_default_validator()?;
+    }
+
+    Ok(())
 }
 
-fn extract_git_hash() -> Result<()> {
-    let mut config = Config::default();
-    *config.git_mut().branch_mut() = false;
-    *config.git_mut().commit_timestamp_mut() = false;
-    *config.git_mut().semver_mut() = false;
-    vergen(config).map_err(|err| eyre!(Box::new(err)))
+fn extract_git_hash() -> Result<(), Box<dyn Error>> {
+    vergen::EmitBuilder::builder()
+        .git_sha(true)
+        .cargo_features()
+        .emit()?;
```

Comment:
Here, `wrap_err` would have been useful.

Reply:
Alternatively, there's only one fallible operation, I don't see why the concrete error type couldn't be used.